modules/ui/sections/map_daterange.js

```diff
@@ -0,0 +1,167 @@
+import { dispatch as d3_dispatch } from 'd3-dispatch';
+import { select as d3_select } from 'd3-selection';
+
+import { t } from '../../core/localizer';
+import { uiTooltip } from '../tooltip';
+import { uiSection } from '../section';
+import { utilQsString, utilStringQs } from '../../util';
+
+const DEFAULT_MIN_DATE = '-4000-01-01';
+const DEFAULT_MAX_DATE = (new Date()).getFullYear() + '-12-31';
+
+const INPUT_STYLES = [
+    { name: 'width', value: '125px' },
+    { name: 'text-align', value: 'center' },
+];
+const LABEL_STYLES = [
+    { name: 'font-weight', value: 'bold' },
+    { name: 'display', value: 'inline-block' },
+    { name: 'width', value: '75px' },
+];
+
+export function uiSectionDateRange(context) {
+    // despite appearing as a separate panel, Map Features does the real filtering
+    // see applyDateRange() in this panel, where the dateRange value is set
+    // see modules/renderer/features.js checkDateFilter() which applies the filters
+    // see modules/renderer/features.js update() which updates URL params
+
+    const section = uiSection('date_ranges', context)
+        .label(t('date_ranges.title'))
+        .disclosureContent(renderDisclosureContent)
+        .expandedByDefault(false);
```

Comment:
In the future, maybe we could consider replacing this separate section with a selectively disclosed row right under the “OpenHistoricalMap Data” checkbox in the Data Layers section. This would make the date filter more discoverable and also consistent with the [street-level imagery date filter](https://github.com/openstreetmap/iD/blob/e9bdd79887c5c8272c3ac9d345aa10943adaf275/modules/ui/sections/photo_overlays.js#L214-L231).